python/modules/IcePy/Util.cpp

```diff
@@ -223,15 +224,33 @@ IcePy::getFunction()
 
 IcePy::PyObjectHandle::PyObjectHandle(PyObject* p) : _p(p) {}
 
-IcePy::PyObjectHandle::PyObjectHandle(const PyObjectHandle& p) : _p(p._p) { Py_XINCREF(_p); }
+IcePy::PyObjectHandle::PyObjectHandle(const PyObjectHandle& p) : _p(p._p)
+{
+    AdoptThread adoptThread; // Ensure that we can call Python code.
```

Comment:
I am not fond of this code. It suggests we're often creating PyObjectHandle from "C++ threads" but I doubt it's common.

The AdoptThread should be in the caller, not here.